src/Evolution/Systems/CurvedScalarWave/Worldtube/Triggers/OrbitRadius.hpp

```diff
@@ -0,0 +1,68 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <pup.h>
+#include <vector>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "Time/Tags/TimeStep.hpp"
+#include "Time/Time.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+
+namespace Triggers {
+/*!
+ * \brief This trigger returns true when the scalar charge is about to cross one
+ * of the specified areal radii.
+ *
+ * \details As the domain is adjusted to track the position of the scalar
+ * charge, the time step needs to be dynamically adjusted accordingly. This
+ * trigger can be used to set the time step according to the radial position of
+ * the scalar charge which gives a good approximation of when the time step
+ * should be adjusted.
+ * The trigger only approximates whether the particle might cross during the
+ * next time step and may therefore fire twice.
+ */
+class OrbitRadius : public Trigger {
+ public:
+  /// \cond
+  OrbitRadius() = default;
+  explicit OrbitRadius(CkMigrateMessage* /*unused*/) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(OrbitRadius);  // NOLINT
+  /// \endcond
+
+  struct OptionTags {
+    struct Radii {
```

Comment:
Any reason why `Radii` needs to be within another struct? Otherwise you can just remove `OptionTags`.